socorro/tests/external/es/test_crashstorage.py

```diff
@@ -378,16 +333,111 @@ def test_index_data_capture(self):
+            pytest.param(
+                "processed_crash.background_task_name",
+                "a" * 10_001,
+                "a" * 10_000,
+                id="long_keywords_truncated",
+            ),
+            # Non-string string converted to "BAD DATA"
+            ("processed_crash.user_comments", 44, "BAD DATA"),
+            # Long strings are truncated
+            pytest.param(
+                "processed_crash.user_comments",
+                "a" * 32_767,
+                "a" * 32_766,
+                id="long_strings_truncated",
+            ),
+            # Out-of-bounds integers are removed
+            (
+                "processed_crash.mac_available_memory_sysctl",
+                2_147_483_999,
+                REMOVED_VALUE,
+            ),
+            # Out-of-bounds longs are removed
+            (
+                "processed_crash.memory_explicit",
+                9_223_372_036_854_775_999,
+                REMOVED_VALUE,
+            ),
+            # Booleans are converted
+            # FIXME(willkg): fix_boolean is never called--that's wrong
+            # ("processed_crash.accessibility", "true", True),
+        ],
+    )
+    def test_indexing_bad_data(self, key, value, expected_value, es_helper):
+        crash_id = create_new_ooid()
+        raw_crash = {"ProductName": "Firefox", "ReleaseChannel": "nightly"}
+        processed_crash = {
+            "date_processed": date_from_ooid(crash_id),
+            "uuid": crash_id,
+        }
+
+        # Put the processed_crash in a doc like we index so the keys match the shape of
+        # the structure, then set the value, then extract the processed crash structure
+        # so we can save it
+        doc = {"processed_crash": processed_crash}
+        glom.assign(doc, key, value, missing=dict)
+        processed_crash = doc["processed_crash"]
+
+        # Save the crash data and then fetch it and verify the value is as expected
+        crashstorage = self.build_crashstorage()
+        crashstorage.save_processed_crash(
+            raw_crash=raw_crash,
+            processed_crash=processed_crash,
+        )
+        es_helper.refresh()
+
+        doc = es_helper.get_crash_data(crash_id)
+        assert glom.glom(doc, key, default=REMOVED_VALUE) == expected_value
+
+    # FIXME(willkg): write tests for index error handling; this is tricky because
+    # when we have index error handling, we write code to fix the error, so it's
+    # hard to test the error situation
+
 
 @pytest.mark.parametrize(
     "value, expected",
     [
         # Non-string values are converted to "BAD DATA"
         (1, "BAD DATA"),
-        # String values are truncated if > 10,000 characters
-        ("a" * 9_999, "a" * 9_999),
-        ("a" * 10_000, "a" * 10_000),
-        ("a" * 10_001, "a" * 10_000),
+        # Keyword string values are truncated if > 10,000 characters
+        pytest.param("a" * 9_999, "a" * 9_999, id="keyword_9_999"),
```

Comment:
I learned about using `pytest.param` to specify the test id and it fixes these situations where an id with 9,999 characters is unusable.